src/Storages/ObjectStorage/DataLakes/IcebergMetadata.h

```diff
@@ -89,7 +89,7 @@ class IcebergMetadata : public IDataLakeMetadata, private WithContext
     bool operator ==(const IDataLakeMetadata & other) const override
     {
         const auto * iceberg_metadata = dynamic_cast<const IcebergMetadata *>(&other);
-        return iceberg_metadata && getVersion() == iceberg_metadata->getVersion();
+        return iceberg_metadata && getManifestListFile() == iceberg_metadata->getManifestListFile();
```

Comment:
I do not understand this change, tbh. I think that we shouldn't say that two metadata objects are equal if they have the same manifest lists but different table schema or partition schema

Reply:
I can use a tuple or something and also compare the schema and partition schema.

The problem with the code as it existed is that it only compared the metadata version, but the top-level metadata contains a history of many snapshots, schemas, etc. And so if you try to query an older version (via timestamp) it would do this comparison, decide the older snapshot you're trying to query is the same as the current snapshot, [and then noop](https://github.com/ClickHouse/ClickHouse/blob/e3bb7a38f1efce03deafbab3103d4b329c9ea50d/src/Storages/ObjectStorage/DataLakes/IStorageDataLake.h#L100-L102).

Reply:
Using a tuple of `metadata_version, schema_id, snapshot_id` now, although in some sense I think I could just use `snapshot_id` since it implies the rest. 🤷